pelican/generators.py

```diff
@@ -175,28 +175,36 @@ def get_files(self, paths, exclude=[], extensions=None):
                 files.append(path)  # can't walk non-directories
         return files
 
-    def add_source_path(self, content):
+    def add_source_path(self, content, static=False):
         """Record a source file path that a Generator found and processed.
         Store a reference to its Content object, for url lookups later.
         """
         location = content.get_relative_source_path()
-        self.context['filenames'][location] = content
+        key = '{}_content'.format('static' if static else 'generated')
```

Comment:
This is a bit nit-picking but I find this much more readable:
```python
key = 'static_content' if static else 'generated_content'
```

Reply:
I agree...